### PR TITLE
feat(organizations): add organization switcher to the dashboard shell

### LINKED ISSUE #7664: Feature: Add organization switcher to dashboard header
### Is your feature request related to a problem? Please describe

Users who belong to multiple organizations have no quick way to switch between them while inside an organization's dashboard. To move to a different org, they currently have to navigate back to the organizations list and find it again.

### Describe the solution you'd like

Add a "Switch" dropdown in the organization dashboard header (next to the organization name) that lets the user jump between the organizations they belong to.

Each entry in the dropdown shows:
- The organization's logo, or an initials badge as a fallback when no logo is set
- The user's role in that organization (Admin / Mentor / Member)
- The number of groups in that organization
- A checkmark on the organization currently being viewed

A "Create organization" link appears at the bottom of the dropdown.

**Implementation notes:**
- Lives in the dashboard header, so it appears across all org tabs (it's part of the shared `OrganizationDashboardShellComponent`).
- Uses the current user's existing memberships as the data source — no new controller actions required.
- Uses a Bootstrap 5 dropdown for the open/close behavior (no custom JS).
- The user's role per org is read from the `organization_members` join, and each org's group count is included alongside it.

### Describe alternatives you've considered

N/A

### Additional context

Project 6 - Part of the Enterprise & Institutional Organization Features

### Are you working on this?

Yes

### PR DESCRIPTION
**Pull request by naman79820**
Fixes #7664 

 

#### Describe the changes you have made in this PR -
Adds an organization switcher dropdown to the organization dashboard header, letting users switch between the organizations they belong to.

- Added a `set_user_organizations` before_action in `OrganizationsController` that gathers the current user's organizations along with their role and group count in each.
- Extended `OrganizationDashboardShellComponent` to render the switcher: added `initials` (initials fallback for orgs without a logo) and `current_org?` (marks the currently-viewed org) helper methods, and the dropdown markup in its template.
- Each entry shows the org's logo (or initials), the user's role, the group count, and a checkmark on the current org, with a "Create organization" link at the bottom.
- Uses a Bootstrap dropdown for open/close behavior
- Added scoped SCSS for the switcher styling.

### Screenshots of the UI changes (If any) -

https://github.com/user-attachments/assets/6a2b2503-4a32-4e49-9493-a112c8759f74

---

## Code Understanding and AI Usage

**Did you use AI assistance (ChatGPT, Claude, Copilot, etc.) to write any part of this code?**
- [x] No, I wrote all the code myself
- [ ] Yes, I used AI assistance (continue below)

**If you used AI assistance:**
- [ ] I have reviewed every single line of the AI-generated code
- [ ] I can explain the purpose and logic of each function/component I added
- [ ] I have tested edge cases and understand how the code handles them
- [ ] I have modified the AI output to follow this project's coding standards and conventions

**Explain your implementation approach:**
- Now user can switch to Organization easily.
- No alternative consideration
- I used components because as per my mentor i shoudn't use partials and i already created components for dashboard title so implemented logic in that file
-  initials function will create a logo with starting two org name if they didn't attach org logo and current_org gets organization id.

---

## Checklist before requesting a review
- [x] I have added proper PR title and linked to the issue
- [x] I have performed a self-review of my code
- [x] **I can explain the purpose of every function, class, and logic block I added**
- [x] I understand why my changes work and have tested them thoroughly
- [x] I have considered potential edge cases and how my code handles them
- [x] If it is a core feature, I have added thorough tests
- [x] My code follows the project's style guidelines and conventions

---

Note: Please check **Allow edits from maintainers** if you would like us to assist in the PR.

## Summary by CodeRabbit

* **New Features**
  * Added an organization switcher to the dashboard header.
  * View organizations with logos or initials, roles, group and member counts, and current-organization status.
  * Select an organization directly from the switcher with clear selection indicators.
  * Added a quick action to create a new organization.
  * Added responsive styling, scrolling, and hover/focus states.
  * Added localized labels for the switcher and its actions.

### CODE REVIEW

**Review by vedant-jain03 — CHANGES_REQUESTED** [review]
We need a limit for list, let's go with 10 and have "Load more" button!

**Comment by naman79820** [comment]
> We need a limit for list, let's go with 10 and have "Load more" button!

Added load more as discussed :))

File: app/controllers/organizations_controller.rb [hunk 1]
```diff
@@ -69,6 +71,15 @@ def check_slug
...
   end
 
+  def switcher_organizations
+    page = params[:page].to_i
+    organizations = switcher_page(page)
+    has_more = current_user.organization_members.count > (page + 1) * SWITCHER_PER_PAGE
+    html = render_to_string(partial: "organizations/switcher_item", collection: organizations, as: :item,
+                            formats: [:html])
+    render json: { html: html, has_more: has_more }
+  end
```
Comment (vedant-jain03) [inline app/controllers/organizations_controller.rb:84]: Make the load more reusable for other, we may opt for any gem for it
@tachyons if you are aware of any

✅ Confirmed as addressed by @naman79820

Comment (tachyons) [reply app/controllers/organizations_controller.rb:84]: @coderabbitai suggest idiomatic rails way to implement this

Comment (tachyons) [reply app/controllers/organizations_controller.rb:84]: @coderabbitai turbo stream uses webscoket, and we prefer not to use it, could you simplify the logic and use stimulas js instead

Comment (tachyons) [reply app/controllers/organizations_controller.rb]: @coderabbitai instead responding html, backend can respond with json and stimulus should be able to render it properly, right ? Or reload the whole page with page parameter ?

Comment (tachyons) [reply app/controllers/organizations_controller.rb:84]: @coderabbitai instead responding html, backend can respond with json and stimulus should be able to render it properly, right ? Or reload the whole page with page parameter ?

**Review by PRATHAM2002-DS — APPROVED** [review]

File: app/controllers/organizations_controller.rb (hunk 1 above)
Comment (naman79820) [reply app/controllers/organizations_controller.rb:84]: Hey @coderabbitai, I've made the changes you suggested. I went with the JSON response containing server rendered HTML as you recommended, and moved the fetch and pagination logic into a Stimulus controller.

Also one more thing, the controller is named load_more_controller.js rather than switcher-specific, so it can be reused for other paginated lists. This will increase reusability mentioned by my mentors.

Can you now review these changes

Comment (naman79820) [reply app/controllers/organizations_controller.rb:84]: i have done the suggested changes @coderabbitai, is it good overall now?

Comment (naman79820) [reply app/controllers/organizations_controller.rb:84]: I have done the suggested changes.

@tachyons could you please take a look? I've moved the load-more logic into a Stimulus controller as you suggested, and kept the JSON response with server-rendered HTML per CodeRabbit's recommendation. Let me know if this looks good to you.
